### ISSUE TITLE
Doubts related to the training and testing of the supervised model

### ISSUE THREAD

**Issue by avijit9** [jonmun/MM-SADA_Domain_Adaptation_Splits#2]
Hi,

I am trying to replicate the supervised model on domain D1 (both training and testing) mentioned in Table 1 of the paper. Right now, I am training the network for both modalities separately. During testing, I am averaging the logits I am getting from both the streams. I am getting around `~55%` top-1 accuracy instead of `62.8\%`. So, just wanted to check if I am doing things correctly. 

1. Did you train both the model jointly?
2. For RGB stream, the normalization values are same as Imagenet? For Flow, the values are mean and std 0.5?
3. If I use 5 temporal clips during testing, the accuracy goes down further. My guess is my sampling strategy is not good. Is it possible to give me some hints on that?

**Comment by jonmun** [comment]
Hi,

I am sorry you were unable to replicate the results. I am in the process of preparing my code for release.

In the meantime I can answer your questions:
1. The model was trained jointly as specified by the L_y loss in our paper.
2. Pixel values were normalized to values between 0 and 1 then centred around 0. This is the same for both RGB and Flow modalities.
3. We averaged the classification scores (after softmax) from 5 sequences taken equidistant along a video segment.  The 5 centre frames are found using the start and end time of a video segment: centre_frames = np.linspace(segment_start,segment_end,7,dtype=np.int32)[1:-1].
Frames were sampled at 30fps and if the video clip contains less than 16 frames the neighbouring frames were used.

I hope this helps and will share the code as soon is it is ready for release.

**Comment by avijit9** [comment]
Thanks @jonmun  for you reply. I'll try your suggestions now, and let you know.

**Comment by avijit9** [comment]
Hi,

I tried your suggestions. But still no luck. I am not able to identify the problem.

 When do you plan to release your implementations? Is it possible to share the code in the current state at least? 

Thanks

**Comment by avijit9** [comment]
Hi, 

Sorry for troubling you again. Note that, all the questions here are for the supervised model (i.e. train on D_i and test on D_i)

1. I am sampling clips from action segments using the following methods - 
```
`def _sample_indices(self, num_frames):
        if num_frames > self.clip_size:
            start_pos = randint(num_frames - self.clip_size + 1)
            indices = range(start_pos, start_pos + self.clip_size, 1)
        else:
            indices = [x for x in range(num_frames)]
        if len(indices) < self.clip_size:
            self._loop_indices(indices)
        return indices


    def _loop_indices(self, indices):
        indices_cycle = cycle(indices)
        while len(indices) < self.clip_size:
            indices.append(next(indices_cycle))

    def _sample_indices_test(self, num_frames):
        #centers = np.linspace(0, num_frames, 7, dtype=np.int32)[1:-1]
        centers = [int(idx) for idx in np.linspace(self.clip_size//2, num_frames-self.clip_size//2, 5)]

        all_indices = []
        
        for c in centers:
            start_pos = max(0, c - self.clip_size // 2)
            end_pos = min(num_frames, c + self.clip_size // 2)
            indices = list(range(start_pos, end_pos, 1))
            
            if len(indices) < self.clip_size:
                self._loop_indices(indices)
            all_indices.extend(indices)
            
        return all_indices
`
```

Is it the identical as you did?

2. Also, I am normalizing by dividing the pixels using 255 and subtracting mean 0.5 and dividing them by std 0.5. Is it right?
3. Did you use pretrained-weights for I3D trained on Imagenet + Kinetics?
4. What is the learning rate did you use? Is it the same as written in the paper?
